src/components/themed/TransactionListTop.tsx

```diff
@@ -493,7 +493,7 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
     const styles = getStyles(theme)
     const { countryCode } = this.state
 
-    const showStaking = this.isStakingAvailable()
+    const hideStaking = !isStakingSupported(this.props.wallet.currencyInfo.pluginId, this.props.currencyCode)
```

Comment:
There's a remaining corner case that any asset with the currency code FIO (like a custom token) will show the earn button and navigate to actual FIO staking. The FIO plugin special currency info has `isStakingSupported: true` so that special case in the helper function can be removed.